File: CPP/CalculateReverse.cpp
#include <iostream>
#include <cmath>
using namespace std;
/* Take N as input, Calculate it's reverse also Print the reverse. */
int main(int argc, char const *argv[])
{
	unsigned long int n, t, res = 0;
	cin >> n;
	t = n;
	int p = 0;
	while(t){
		t/=10;
		p++;
	}
	p-=1;

	while(n){
		short int rem = n % 10;
		res += rem*pow(10, p--);
		n/=10;
	}
	cout << res << endl;
	return 0;
}

File: Java/CalculateReverse.java
import java.util.Scanner;
/* Take N as input, Calculate it's reverse also Print the reverse. */
class CalculateReverse{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String s = sc.next();
		// 1. count the digits;
		int p = s.length()-1;
		long n = Long.parseUnsignedLong(s);
		long res = 0;
		while(n!=0){
			short rem = (short)(n%10);
			res += rem*Math.pow(10, p--);
			n/=10;
		}
		System.out.println(res);
	}
}